eng/build-native.sh

```diff
@@ -39,6 +40,7 @@ usage()
     echo "Usage: $0 [options]"
     echo "--build-native - build native components"
     echo "--test - test native components"
+    echo "--copy-sos - copy the SOS binaries to dotnet-sos/dotnet-dump"
```

Comment:
When would I not want to do this copy? This feels like a surprisingly specific flag to get surfaced on the build.

Reply:
I guess it is a little overkill. I was trying to prevent copy the binaries twice: at the end of the build and before the tests are run.  The later is needed for the "test only" runs where there is no build.

It isn't going to hurt to copy them everything somebody runs "build.sh" or "test.sh". I'll remove it.